src/plugins/controls/public/range_slider/components/range_slider_control.tsx

```diff
@@ -37,45 +53,132 @@ export const RangeSliderControl: FC = () => {
+   */
   useEffect(() => {
     setDisplayedValue(value ?? ['', '']);
   }, [value]);
 
-  const button = (
-    <RangeSliderButton
-      value={displayedValue}
-      onChange={setDisplayedValue}
-      isPopoverOpen={isPopoverOpen}
-      setIsPopoverOpen={setIsPopoverOpen}
-    />
+  const ticks: EuiRangeTick[] = useMemo(() => {
+    return [
+      { value: min ?? -Infinity, label: fieldFormatter(String(min)) },
+      { value: max ?? Infinity, label: fieldFormatter(String(max)) },
+    ];
+  }, [min, max, fieldFormatter]);
+
+  const levels = useMemo(() => {
+    return [
+      {
+        min: min ?? -Infinity,
+        max: max ?? Infinity,
+        color: 'success',
+      },
+    ];
+  }, [min, max]);
+
+  const disablePopover = useMemo(
+    () =>
+      isLoading ||
+      displayedMin === -Infinity ||
+      displayedMax === Infinity ||
+      displayedMin === displayedMax,
+    [isLoading, displayedMin, displayedMax]
+  );
+
+  const getCommonInputProps = useCallback(
+    ({
+      inputValue,
+      testSubj,
+      placeholder,
+    }: {
+      inputValue: string;
+      testSubj: string;
+      placeholder: string;
+    }) => {
+      return {
+        isInvalid,
+        placeholder,
+        readOnly: false, // overwrites `canOpenPopover` to ensure that the inputs are always clickable
+        className: 'rangeSliderAnchor__fieldNumber',
+        'data-test-subj': `rangeSlider__${testSubj}`,
+        value: inputValue === placeholder ? '' : inputValue,
+      };
+    },
+    [isInvalid]
   );
 
   return error ? (
     <ControlError error={error} />
   ) : (
-    <EuiInputPopover
-      input={button}
-      isOpen={isPopoverOpen}
-      display="block"
-      panelPaddingSize="s"
-      className="rangeSlider__popoverOverride"
-      anchorClassName="rangeSlider__anchorOverride"
-      panelClassName="rangeSlider__panelOverride"
-      closePopover={() => {
-        setIsPopoverOpen(false);
-      }}
-      anchorPosition="downCenter"
-      attachToAnchor={false}
-      disableFocusTrap
-      onPanelResize={(width) => {
-        rangeRef.current?.onResize(width);
-      }}
-    >
-      <RangeSliderPopover rangeRef={rangeRef} value={displayedValue} onChange={setDisplayedValue} />
-    </EuiInputPopover>
+    <span className="rangeSliderAnchor__button" data-test-subj={`range-slider-control-${id}`}>
+      <EuiDualRange
+        ref={rangeSliderRef}
+        id={id}
+        fullWidth
+        showTicks
+        ticks={ticks}
+        levels={levels}
+        min={displayedMin}
+        max={displayedMax}
+        isLoading={isLoading}
+        onMouseUp={() => {
+          // when the pin is dropped (on mouse up), cancel any pending debounced changes and force the change
+          // in value to happen instantly (which, in turn, will re-calculate the min/max for the slider due to
+          // the `useEffect` above.
+          debouncedOnChange.cancel();
```

Comment:
Debouncing this in this way makes interacting with the slider pretty seamless! Nice work!